cpp/src/arrow/util/ubsan.h

```diff
@@ -60,6 +60,14 @@ inline typename std::enable_if<std::is_trivial<T>::value, T>::type SafeLoadAs(
   return ret;
```

Comment:
you could probably make non-length SafeLoadAs call SafeLoadAs with `sizeof(T)` as `length`

Reply:
Fixed